modules/eventbridge/main.tf

```diff
@@ -30,8 +30,8 @@ resource "aws_iam_role_policy_attachment" "firehose" {
 }
 
 resource "aws_cloudwatch_event_target" "firehose" {
-  count    = length(var.rules)
+  for_each = var.rules
```

Comment:
I'd be surprised if this works. `for_each` works with `set` and `map`, but only `set(string)` for the former. Otherwise, it can't know how to uniquely identify each element in the collection. You'd get an error that's something like "`for_each` expression is not suitable."

Reply:
I'd expect you'd need something like this:

```tf
for_each = {
  for r in var.rules : r.id => r
}
```

Reply:
I guess `terraform validate` doesn't catch this, even though this can be determined directly from the type. Not all invalid `for_each` expressions can be detected at validation time, e.g.:

```tf
for_each = jsondecode(data.my_source.json)
```

Valid if `json` is a JSON object, invalid in all other cases.

Reply:
Good catch, here I was thinking `terraform validate` worked.